arkouda/dataframe.py

```diff
@@ -1841,6 +1844,269 @@ def numeric_help(d):
+        regEx = compile(
+            f"^df_data_[a-zA-Z0-9]+_({pdarray.objtype}|{Strings.objtype}|"
+            f"{Categorical.objtype}|{SegArray.objtype})_{user_defined_name}"
+        )
+        # Using the regex, cycle through the registered items and find all the columns in the DataFrame
+        for name in registry:
+            x = match(regEx, name)
+            if x is not None:
+                matches.append(x.group())
+        matches.sort()
+
+        if len(matches) == 0:
+            raise RegistrationError(f"No registered elements with name '{user_defined_name}'")
+
+        # Remove duplicates caused by multiple components in Categorical or SegArray
+        dedupe = list(dict.fromkeys(matches))
+
+        for name in dedupe:
+            colName = name.split("_")[2]
+            if f"_{Strings.objtype}_" in name or f"_{pdarray.objtype}_" in name:
+                cols_resp = cast(str, generic_msg(cmd="attach", args=name))
+                dtype = cols_resp.split()[2]
+                if dtype == Strings.objtype:
+                    columns[colName] = Strings.from_return_msg(cols_resp)
+                else:  # pdarray
+                    columns[colName] = create_pdarray(cols_resp)
+            elif f"_{Categorical.objtype}_" in name:
+                columns[colName] = Categorical.attach(name)
+            elif f"_{SegArray.objtype}_" in name:
+                columns[colName] = SegArray.attach(name)
+
+        index_resp = cast(str, generic_msg(cmd="attach", args=f"df_index_{user_defined_name}_key"))
+        dtype = index_resp.split()[2]
+        if dtype == Strings.objtype:
+            ind = Strings.from_return_msg(index_resp)
+        else:  # pdarray
+            ind = create_pdarray(index_resp)
+
+        index = Index.factory(ind)
+
+        df = DataFrame(columns, index)
+        df.name = user_defined_name
+        return df
+
+    @staticmethod
+    @typechecked
+    def unregister_dataframe_by_name(user_defined_name: str) -> None:
+        """
+        Function to unregister DataFrame object by name which was registered
+        with the arkouda server via register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            Name under which the DataFrame object was registered
+
+        Raises
+        -------
+        TypeError
+            if user_defined_name is not a string
+        RegistrationError
+            if there is an issue attempting to unregister any underlying components
+
+        See Also
+        --------
+        register, unregister, attach, is_registered
+        """
+
+        registry = list_registry()
+        matches = []
+        regEx = compile(
+            f"^df_data_[a-zA-Z0-9]+_({pdarray.objtype}|{Strings.objtype}|"
+            f"{Categorical.objtype}|{SegArray.objtype})_{user_defined_name}"
+        )
+        # Using the regex, cycle through the registered items and find all the columns in the DataFrame
+        for name in registry:
```

Comment:
could we just do `for name in registry` and then remove ln 2072?

Reply:
Updated